File: appointly-api/Appointly.Domain/Entities/Patient.cs
using Appointly.Domain.Common;
using Appointly.Domain.Enums;

namespace Appointly.Domain.Entities;

public class Patient : BaseModel
{
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public DateTime DateOfBirth { get; set; }
    public string PhoneNumber { get; set; }
    public string Email { get; set; }
    public decimal Fee { get; set; }
    public bool IsSpecialPatient { get; set; }
    public bool HasAResponsible { get; set; }
    
    // criar entidade reposponsible
    public string? ResponsibleName { get; set; }
    public string? ResponsibleEmail { get; set; }
    public string? ResponsiblePhoneNumber { get; set; }
    public RelationshipDegree? RelationshipDegree { get; set; }
    
    public List<Appointment> Appointments { get; set; }
}

File: appointly-app/src/views/patients/patient.ts
export enum RelationshipDegreeEnum {
  PARENT = 'PARENT',
  CHILD = 'CHILD',
  SIBLING = 'SIBLING',
  SPOUSE = 'SPOUSE',
  GRANDPARENT = 'GRANDPARENT',
  GRANDCHILD = 'GRANDCHILD',
  AUNT_UNCLE = 'AUNT_UNCLE',
  NIECE_NEPHEW = 'NIECE_NEPHEW',
  COUSIN = 'COUSIN',
}

export type Patient = {
  id?: string
  firstName: string
  lastName: string
  dateOfBirth: string
  phoneNumber: string
  email: string
  fee: number
  isSpecialPatient: boolean
  hasAResponsible: boolean
  responsibleName?: string | null
  responsibleEmail?: string | null
  responsiblePhoneNumber?: string | null
  relationshipDegree?: RelationshipDegreeEnum | null
}
